src/ConstructiveSolidGeometry/IO.jl

```diff
@@ -188,9 +197,11 @@ end
 # end
 
 
-function Geometry(::Type{T}, dict::AbstractDict, input_units::NamedTuple) where {T}
-    key::String, transformations = get_geometry_key(T, dict, input_units)
-    transform(Geometry(T, CSG_dict[key], dict[key], input_units), transformations)
+function Geometry(::Type{T}, dict::AbstractDict, input_units::NamedTuple, outer_transformations) where {T}
+    key = get_geometry_key(T, dict, input_units)
+    inner_transformations = parse_CSG_transformation(T, dict, input_units)
+    transformations = combine_transformations(inner_transformations, outer_transformations)
```

Comment:
Ìs there a more efficient way than constructing a new `NamedTuple` each time `Geometry` is called?

Reply:
NamedTuples are (in this case) fully typed. So it should be very efficient.

Reply:
One could change it into one line and just have the variable `transformation`. But it won't be that clear to a human read. 
And would make no difference for the compiler.

Reply:
Is there a way to update `outer_transformations` and then call something like
`add!(transformations, inner_transformations)`.
The types of the transformations should not change.

Reply:
This is not needed here. The compiler takes care about this automatically as `Transformation{T}` is fully typed.